lib/external/io.dart

```diff
@@ -23,11 +26,38 @@ ASTNode funcFree(Runtime runtime, ASTNode self, List<ASTNode> args) {
   final Scope scope = getScope(runtime, self);
 
   args.forEach((arg) {
-    if (arg is! VariableNode)
-      return;
+    if (arg is! VariableNode) return;
 
-    scope.variableDefinitions.removeWhere((varDef) => varDef.variableName == arg.variableName);
+    scope.variableDefinitions
+        .removeWhere((varDef) => varDef.variableName == arg.variableName);
   });
 
   return INITIALIZED_NOOP;
-}
\ No newline at end of file
+}
+
+Future<MapNode> funcSystem(
+    Runtime runtime, ASTNode self, List<ASTNode> args) async {
+  runtimeExpectArgs(
+      args, [ASTType.AST_STRING, ASTType.AST_LIST, ASTType.AST_BOOL]);
+
+  Map<String, dynamic> result = {
+    'errorCode': null,
+    'stdout': null,
+    'stderr': null
+  };
+
+  MapNode res = MapNode();
+  bool shell = args[2].boolVal ?? false;
+
+  final ProcessResult pr = await Process.run(args[0].stringValue,
+      args[1].listElements.map((e) => e.toString()).toList(),
+      runInShell: shell);
+
+  result['errorCode'] = pr.exitCode;
+  result['stdout'] = pr.stdout;
+  result['sterr'] = pr.stderr;
```

Comment:
This exposes the value (List<int> or String) as dart types, I think they should be wrapped in ASTNode. It probably will error in birb if any kind of operation is done on it. @Calamity210 thoughts?

Reply:
They are converted see the last line. Line 60

Reply:
I cant recall completely how maps are handled off the top of my head, but i dont think it would cause an issue.